src/contractIntegration/interaction/web3Provider/FactoryWeb3Interaction.ts

```diff
@@ -0,0 +1,62 @@
+import 'reflect-metadata';
+import { container } from 'tsyringe';
+
+import IInteractionConfig from "../IInteractionConfig";
+import IOption from "../IOption";
+import IWeb3Provider from './IWeb3Provider';
+import Web3ProviderClam from "./Web3ProviderClam";
+import Web3ProviderBowhead from "./Web3ProviderBowhead";
+
+
+class FactoryWeb3Interaction {
+    private config: IInteractionConfig;
+    private web3Providers: IOption[];
+    public static intance: FactoryWeb3Interaction;
+
+    private constructor() {
+        this.web3Providers = [
+            { name: "web3", option: Web3ProviderClam },
+            { name: "bowhead", option: Web3ProviderBowhead }
+        ];
+    };
+
+    public static getInstance(): FactoryWeb3Interaction {
+        if (!FactoryWeb3Interaction.intance) {
+            this.intance = new FactoryWeb3Interaction();
+        }
+        return this.intance;
+    }
+    
+    public setConfig(config: IInteractionConfig): void {
+        this.config = config;
+    }
+
+    public setOptionWeb3Provider(option: IOption): void {
+        if (option.name.trim() === '' && option.name.trim().length === 0) {
+            throw new Error('The name must have at least one character');
+        }
+        const optionWeb3Provider = this.web3Providers.find(optionAux => optionAux.name === option.name);
+        if (optionWeb3Provider) {
+            throw new Error('This option already exists.');
```

Comment:
@HectorGerardoZV Allow devs to update an existing option

Reply:
....
setOptionWeb3Provider({name:"urlProvider", value:"http://localhost:8080"})

...
// later in the code
setOptionWeb3Provider({name:"urlProvider", value:"http://rinkeby.com"})